examples/derive_ref/README.md

```diff
@@ -224,6 +224,7 @@ Notes:
 - `from_occurrences`:
   - Implies `arg.takes_value(false).multiple_occurrences(true)`
   - Reads from `clap::ArgMatches::occurrences_of` rather than a `value_of` function
+  - Does not work when `default_value_t` is also specified
```

Comment:
"does nothing" and "does not work" are both inaccurate descriptions.  We should be providing the building blocks for understanding rather than providing terminal statements.

`default_value` implies `takes_value(true)` which will override the `takes_value(false)` from `from_occurrences.  The user can revert things back.  However, `default_value` implies its for the value and we've documented `takes_value(false)` in this case.  In particular, we do not consider defaults to have "occurred", so even with the below working code, `default_value_t` (or any other default attribute) will be ignored
```rust
use clap::{IntoApp, Parser};

#[derive(Parser, Debug)]
struct MyCommandOptions {
    #[clap(
        short,
        long,
        parse(from_occurrences),
        default_value_t = 1,
        takes_value = false
    )]
    verbose: usize,
}

fn main() {
    let x = MyCommandOptions::parse();
    dbg!(x);
    // code ...
}
```

My initial suggestion would be: `Defaulting (e.g. `default_value_t`) is unlikely to do what you are looking for as this is measuring occurrences and defaults never occurred on the command line.`

That I think heads it off enough that we don't have to get into the minutia of `takes_value`.

Reply:
> My initial suggestion would be: 

how about: `Because "from_occurrences" requires "takes_value(false)", it makes it incompatible with Defaulting (e.g. "default_value_t").`

Reply:
While this is an improvement over what currently exists, it isn't over what has been discussed.  It lacks the explanation for the underlying mechanisms that helps provide insight to users.

I'm going to go ahead and push an update that takes into account both views.